### PR TITLE
Adding Quectel Modems EC25 and BG96

### PR DESCRIPTION
**Pull request by ftoure67**
Added support for Quectel EC25 and BG96.
Quectel EG25 already exists. 
Update in kura.net.admin.modem and kura.linux.net.

**Related Issue:** This PR fixes/closes {issue number}
Issue 2598.

### CODE REVIEW

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/SupportedUsbModemInfo.java [hunk 1]
```diff
@@ -53,9 +54,15 @@
             ModemTechnologyType.UMTS), Arrays.asList(new OptionModemDriver("19d2", "1476")), ""),
     SimTech_SIM7000("SIM7000", "1e0e", "9001", 5, 0, 3, 2, 3, 5000, 10000, Arrays.asList(ModemTechnologyType.LTE,
             ModemTechnologyType.GSM_GPRS), Arrays.asList(new OptionModemDriver("1e0e", "9001")), ""),
-    QUECTEL_EG25("EG25", "2c7c", "0125", 4, 0, 2, 3, -1, 5000, 10000, Arrays.asList(ModemTechnologyType.LTE,
+    QUECTEL_EG25("EG25", "2c7c", "0125", 4, 0, 3, 3, 1, 5000, 10000, Arrays.asList(ModemTechnologyType.LTE,
```
Comment (MMaiero) [inline kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/SupportedUsbModemInfo.java]: Hi, are you sure that the Quectel EG 25 has gps support? Are the AT and data port the same?

File: kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/bg96/QuectelBG96ConfigGenerator.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        pppPeer.setActiveFilter(modemConfig.getActiveFilter());
+        pppPeer.setLcpEchoInterval(modemConfig.getLcpEchoInterval());
+        pppPeer.setLcpEchoFailure(modemConfig.getLcpEchoFailure());
+
+        return pppPeer;
+    }
+
+    @Override
+    public ModemXchangeScript getConnectScript(ModemConfig modemConfig) {
+        int pdpPid = 1;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/bg96/QuectelBG96ConfigGenerator.java]: Is the pdpPid fixed to 1?
What happens if this value is set to 1 and the customer sets a connection string with a different pop context?

File: kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/ec25/QuectelEC25ConfigGenerator.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        pppPeer.setActiveFilter(modemConfig.getActiveFilter());
+        pppPeer.setLcpEchoInterval(modemConfig.getLcpEchoInterval());
+        pppPeer.setLcpEchoFailure(modemConfig.getLcpEchoFailure());
+
+        return pppPeer;
+    }
+
+    @Override
+    public ModemXchangeScript getConnectScript(ModemConfig modemConfig) {
+        int pdpPid = 1;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/ec25/QuectelEC25ConfigGenerator.java]: Same here

File: kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/zte/me3630/ZteMe3630ConfigGenerator.java [hunk 4]
```diff
@@ -72,6 +72,7 @@ public PppPeer getPppPeer(String deviceId, ModemConfig modemConfig, String logFi
 
     @Override
     public ModemXchangeScript getConnectScript(ModemConfig modemConfig) {
+        // PDP context is fixed to 1 for this modem
```
Comment (MMaiero) [inline kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/zte/me3630/ZteMe3630ConfigGenerator.java:75]: Maybe this needs a separate PR as this regards the Quectel modems

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/SupportedUsbModemInfo.java (hunk 1 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/SupportedUsbModemInfo.java]: OK. The modem supports GNSS, but it's not supported yet in the code. I will revert this change.

File: kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/zte/me3630/ZteMe3630ConfigGenerator.java (hunk 4 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/zte/me3630/ZteMe3630ConfigGenerator.java:75]: OK.

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/SupportedUsbModemInfo.java (hunk 1 above)
Comment (sterwen) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/SupportedUsbModemInfo.java]: In fact EC25 and EG25 are identical from a host standpoint, so we will merge both into a single name: Ex_25

**Comment by MMaiero** [comment]
@ftoure67 Is the PR complete?

**Comment by ftoure67** [comment]
Yes. PR is complete.

**Review by pierantoniomerlino — APPROVED** [review]

**Pull request merged**